tests/unit/metadata/test_metadata.py

```diff
@@ -23,7 +23,7 @@ def test_dist_get_direct_url_no_metadata(mock_read_text: mock.Mock) -> None:
     class FakeDistribution(BaseDistribution):
         pass
 
-    dist = FakeDistribution()
+    dist = FakeDistribution()  # type: ignore[abstract]
```

Comment:
Maybe we should make FakeDistribution not abstract instead? We can just raise in all methods that are not supposed to be called.